include/libhal-expander/tla2528.hpp

```diff
@@ -0,0 +1,178 @@
+#pragma once
+#include <libhal/i2c.hpp>
+#include <libhal/steady_clock.hpp>
+#include <libhal/units.hpp>
+
+namespace hal::expander {
+
+// adapters
+class tla2528_adc;
+class tla2528_input_pin;
+class tla2528_output_pin;
+
+/**
+ * @brief tla2528 is a gpio expander & adc mux driver
+ *
+ * tla2528 has 8 pins which can be independently operated as an adc,
+ * digital input, and digital out over i2c. The i2c address is configured by
```

Comment:
```suggestion
 * input pin, and digital pin over i2c. The i2c address is configured by
```

Might as well use the same terminology we use for the interfaces.

Reply:
Which interface are you referring to?

Reply:
`hal::input_pin` and `hal::output_pin` so you can use the terms input pin and output pin which assume they are digital.

Reply:
Updated, look over it over

Reply:
marking as resolved based on general comment
"Looks good! One last little fix in the form of a C style cast and we can submit this."